File: internal/domain/bank.go
package domain

import "time"

// BankTransaction represents an imported bank transaction.
type BankTransaction struct {
	ID                  int64     `json:"id"`
	BankAccount         string    `json:"bank_account"`
	TransactionDate     time.Time `json:"transaction_date"`
	Amount              Amount    `json:"amount"`
	Currency            string    `json:"currency"`
	CounterpartyAccount string    `json:"counterparty_account"`
	CounterpartyName    string    `json:"counterparty_name"`
	VariableSymbol      string    `json:"variable_symbol"`
	ConstantSymbol      string    `json:"constant_symbol"`
	SpecificSymbol      string    `json:"specific_symbol"`
	Message             string    `json:"message"`
	InvoiceID           *int64    `json:"invoice_id,omitempty"`
	Matched             bool      `json:"matched"`
	CreatedAt           time.Time `json:"created_at"`
}

File: rust/crates/zfaktury-domain/src/bank.rs
use chrono::{NaiveDate, NaiveDateTime};

use crate::amount::Amount;

/// An imported bank transaction.
#[derive(Debug, Clone)]
pub struct BankTransaction {
    pub id: i64,
    pub bank_account: String,
    pub transaction_date: NaiveDate,
    pub amount: Amount,
    pub currency: String,
    pub counterparty_account: String,
    pub counterparty_name: String,
    pub variable_symbol: String,
    pub constant_symbol: String,
    pub specific_symbol: String,
    pub message: String,
    pub invoice_id: Option<i64>,
    pub matched: bool,
    pub created_at: NaiveDateTime,
}
